pyquantms/psm/psm_conversion.py

```diff
@@ -0,0 +1,172 @@
+from pathlib import Path
+
+_parquet_field = [
+    "sequence",
+    "protein_accessions",
+    "protein_start_positions",
+    "protein_end_positions",
+    "modifications",
+    "retention_time",
+    "charge",
+    "calc_mass_to_charge",
+    "reference_file_name",
+    "scan_number",
+    "peptidoform",
+    "posterior_error_probability",
+    "global_qvalue",
+    "is_decoy",
+    "consensus_support",
+    "mz_array",
+    "intensity_array",
+    "num_peaks",
+    "search_engines",
+    "id_scores",
+    "hit_rank",
+]
+
+
+def mods_position(peptide):
+    pattern = re.compile(r"\((.*?)\)")
+    original_mods = pattern.findall(peptide)
+    peptide = re.sub(r"\(.*?\)", ".", peptide)
+    position = [i.start() for i in re.finditer(r"\.", peptide)]
+    for j in range(1, len(position)):
+        position[j] -= j
+
+    for k in range(0, len(original_mods)):
+        original_mods[k] = str(position[k]) + "-" + original_mods[k]
+
+    original_mods = (
+        [str(i) for i in original_mods] if len(original_mods) > 0 else np.nan
+    )
+
+    return original_mods
+
+
+@click.command("convert_psm")
+@click.option("--idxml", type=click.Path(exists=True))
+@click.option("--spectra_file", type=click.Path(exists=True))
+@click.option("--export_decoy_psm", is_flag=True)
+@click.pass_context
+def convert_psm(ctx, idxml: str, spectra_file: str, export_decoy_psm: bool = False):
+    """
+    Convert idXML to csv file with PSMs information.
+    :param ctx: click context
+    :param idxml: Input idXML file
+    :param spectra_file: Spectra file
+    :param export_decoy_psm: Export decoy PSM
+    :return: None
+    """
+
+    prot_ids = []
+    pep_ids = []
+    parquet_data = []
+    consensus_support = np.nan
+    mz_array = []
+    intensity_array = []
+    num_peaks = np.nan
+    id_scores = []
+    search_engines = []
+
+    oms.IdXMLFile().load(idxml, prot_ids, pep_ids)
+    if "ConsensusID" in prot_ids[0].getSearchEngine():
+        if prot_ids[0].getSearchParameters().metaValueExists("SE:MS-GF+"):
+            search_engines = ["MS-GF+"]
+        if prot_ids[0].getSearchParameters().metaValueExists("SE:Comet"):
+            search_engines.append("Comet")
+        if prot_ids[0].getSearchParameters().metaValueExists("SE:Sage"):
+            search_engines.append("Sage")
+    else:
+        search_engines = [prot_ids[0].getSearchEngine()]
+
+    reference_file_name = os.path.splitext(
+        prot_ids[0].getMetaValue("spectra_data")[0].decode("UTF-8")
+    )[0]
+    spectra_df = pd.read_csv(spectra_file) if spectra_file else None
+
+    for peptide_id in pep_ids:
+        retention_time = peptide_id.getRT()
+        calc_mass_to_charge = peptide_id.getMZ()
+        scan_number = int(
+            re.findall(
+                r"(spectrum|scan)=(\d+)", peptide_id.getMetaValue("spectrum_reference")
+            )[0][1]
+        )
+
+        if isinstance(spectra_df, pd.DataFrame):
+            spectra = spectra_df[spectra_df["scan"] == scan_number]
```

Comment:
@daichengxin In this line we try to read the scan from the tsv but in the tsv structure we don't have scan, see generated examples from mzml_statistics https://github.com/bigbio/pyquantms/blob/dev/tests/test_data/BSA1_F1_ms_info.tsv